### ISSUE TITLE
Support combo box input type

### ISSUE THREAD

**Issue by nonprofittechy** [jhpyle/docassemble#33]
It would be useful in a lot of circumstances to allow selection from a dropdown menu with the easy possibility to add a new item to the list, such as Windows combobox form element. This isn't supported cross-browser as a standard HTML element but there are extensions to boostrap that handle it, e.g.: https://github.com/danielfarrell/bootstrap-combobox .

This should be an option that can be toggled on or off in the YAML interview file.

**Comment by jhpyle** [comment]
I can see the advantage of combo boxes.  In https://litigationdatabase.org, I spent a lot of time developing a combo box with autocomplete and Ajax retrieval, etc.

But I can think of reasons why they would be a Bad Idea:

1. Users aren't used to them.  My hypothesis is that a big part of usability is doing things the ways that users see every day.
2. Cross-browser and cross-device support would not be reliable.  E.g., I went to the demo of that Bootstrap Combobox on my mobile phone and the page wasn't responsive.  Maybe it does work fine on mobile, but this is a red flag that the developers don't care about keeping up with the latest mobile trends.

Maybe this can be implemented fairly easily, so I'll look into it.

In the meantime, a `choices` dropdown with an "Other" selection, in combination with `show if` [is an option](https://docassemble.org/docs/fields.html#show%20if).
